Voice/AI Agent - NodeJS/index.js

```diff
@@ -0,0 +1,127 @@
+import * as dotenv from "dotenv";
+import fetch from "node-fetch";
+dotenv.config();
+import express from "express";
+import { RestClient } from "@signalwire/compatibility-api";
+const host = process.env.HOST_APP;
+
+const app = express();
+app.use(express.urlencoded({ extended: true }));
+app.use(express.json());
+
+app.post("/", (req, res) => {
+  const response = new RestClient.LaML.VoiceResponse();
+  const connect = response.connect();
+  const agent = connect.ai({
+    voice: "en-US-Neural2-D",
+  });
+  agent.setPostPromptURL(host + "/summary");
+  agent.prompt(
+    {
+      confidence: 0.4,
+      frequencyPenalty: 0.3,
+    },
+    `You are the CEO's assistant.
+    Your job is to answer phone calls and collect messages or transfer the caller to a human agent. 
+    The CEO is not available now. Offer to take a message or transfer the caller to live agent.
+    Request the caller's name.
+    You are able to transfer a call to support or sales. 
+    Here is a list of known numbers that you can transfer the caller to. Do not ask the caller for these numbers. Just transfer the caller by using the appropriate function. 
+    | Name    | Phone Number    |
+    | --------| --------------- |
+    | Sales   | 12057937849     |
+    | Support | 19379028398     |
+    If the caller would like to leave a message for the CEO, ask for their message. 
+    After collecting the message, do not wait for the user to end the conversation: say goodbye and hang up the call. 
+    Be sure to hang up the call at the end of every conversation.`
+  );
+  agent.postPrompt(`Return a valid anonymous json object by replacing the uppercase placeholders in the following template with the caller's information.
+  {"contact_name": "CONTACT_NAME", "message": "MESSAGE or TRANSFERRED"}`);
+
+  const swaig = agent.swaig();
+
+  const transferFn = swaig.function({ name: "transfer" });
+  transferFn.setPurpose("use this when a request for a transfer is made");
+  transferFn.setArgument(
+    'the 10-digit phone number that matches the name given in the transfer request, in JSON format. Example: {"number": "<number>"}'
+  );
+  transferFn.setWebHookURL(
+    `${host}/function?CallSid=${encodeURIComponent(req.body.CallSid)}`
```

Comment:
adding the host may not be needed anymore (it's been fixed), could you verify?

Reply:
This works on `postPromptURL` without the host in staging. It will not work for the public until that change is available in production. An ask is out to extend this fix to `webHookURL` as well.